rpc/staking.go

```diff
@@ -295,7 +244,82 @@ func (s *PublicStakingService) GetAllValidatorInformationByBlockNumber(
 	}
 
 	// Response output is the same for all versions
-	return s.getAllValidatorInformation(ctx, page, blockNum)
+	res, err := s.getPagedValidatorInformationCached(ctx, page, blockNumber)
+	if err != nil {
+		DoMetricRPCQueryInfo(GetAllValidatorInformationByBlockNumber, FailedNumber)
+		return nil, err
+	}
+	return res, nil
+}
+
+func (s *PublicStakingService) getPagedValidatorInformationCached(ctx context.Context, page int, blockNumber BlockNumber) (interface{}, error) {
+	type cacheKey struct {
+		bn   uint64
+		page int
+	}
+	var bn uint64
+	if blockNumber == LatestBlockNumber {
+		bn = s.hmy.CurrentBlock().NumberU64()
+	} else {
+		bn = uint64(blockNumber)
+	}
+
+	key := cacheKey{bn, page}
```

Comment:
right now, the validator information are changing every block. I believe most calls are fetching the latest data from current block, so not sure how much caching helps here. But after the 64 block batch on block reward logic, it may be ok to cache the validator info for every 64 blocks (basically returning the info with at most 64 block staleness).

Reply:
1. I believe the caching is initially aimed to cache for scenario where multiple staking dashboard users querying for one block at the same time. 
2. The data also contains signed / ToSign blocks as a metric of signing rate. Will block reward change also batch changes of these signing metrics?
2. This can also be contributed by shredding the data structure validatorWrapper. Shall revisit this afterwards.

Reply:
1. sounds good.
2. Yes, the signed Tosign are calculated together with block rewards. So it won't be updated every block soon either.
3. Definitely